sci-libs/fftw/fftw-3.3.5.ebuild

```diff
@@ -0,0 +1,180 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=6
+
+FORTRAN_NEEDED=fortran
+
+inherit flag-o-matic fortran-2 toolchain-funcs versionator multibuild multilib-minimal
+
+DESCRIPTION="Fast C library for the Discrete Fourier Transform"
+HOMEPAGE="http://www.fftw.org/"
+
+if [[ ${PV} = *9999 ]]; then
+	inherit autotools git-2
+	EGIT_REPO_URI="https://github.com/FFTW/fftw3.git"
+	KEYWORDS="~alpha ~amd64 ~arm ~hppa ~ia64 ~ppc ~ppc64 ~sparc ~x86"
+else
+	SRC_URI="http://www.fftw.org/${P}.tar.gz"
+	KEYWORDS="~alpha ~amd64 ~arm ~hppa ~ia64 ~mips ~ppc ~ppc64 ~sparc ~x86 ~amd64-linux ~x86-linux ~ppc-macos ~x64-macos ~x86-macos"
+fi
+
+LICENSE="GPL-2+"
+SLOT="3.0/3"
+IUSE="altivec cpu_flags_x86_avx doc cpu_flags_x86_fma3 cpu_flags_x86_fma4 fortran mpi neon openmp quad cpu_flags_x86_sse cpu_flags_x86_sse2 static-libs test threads zbus"
+
+RDEPEND="
+	mpi? ( virtual/mpi )
+	abi_x86_32? ( !<=app-emulation/emul-linux-x86-soundlibs-20130224-r2
+					!app-emulation/emul-linux-x86-soundlibs[-abi_x86_32(-)] )"
+DEPEND="${RDEPEND}
+	test? ( dev-lang/perl )"
+
+pkg_setup() {
+	if [[ ${MERGE_TYPE} != binary ]] && use openmp; then
+		if tc-is-gcc && ! tc-has-openmp; then
+			ewarn "OpenMP is not available in your current selected gcc"
+			die "need openmp capable gcc"
+		fi
+		FORTRAN_NEED_OPENMP=1
+	fi
+
+	fortran-2_pkg_setup
+
+	MULTIBUILD_VARIANTS=( single double longdouble )
+	if use quad; then
+		if tc-is-gcc && ! version_is_at_least 4.6 $(gcc-version); then
+			ewarn "quad precision only available for gcc >= 4.6"
+			die "need quad precision capable gcc"
+		fi
+		MULTIBUILD_VARIANTS+=( quad )
+	fi
+}
+
+src_prepare() {
+	default
+
+	# fix info file for category directory
+	if [[ ${PV} = *9999 ]]; then
+		sed -i -e
+			's/Texinfo documentation system/Libraries/' \
+			doc/fftw3."info" || die "failed to fix info file"
+
+		eautoreconf
+	fi
+}
+
+multilib_src_configure() {
+	# there is no abi_x86_32 port of virtual/mpi right now
+	local enable_mpi=$(use_enable mpi)
+	multilib_is_native_abi || enable_mpi="--disable-mpi"
+
+	# jlec reported USE=quad on abi_x86_32 has too few registers
+	# stub Makefiles
+	if [[ ${MULTILIB_ABI_FLAG} == abi_x86_32 && ${MULTIBUILD_ID} == quad-* ]]; then
+		mkdir -p "${BUILD_DIR}/tests" || die
+		echo "all: ;" > "${BUILD_DIR}/Makefile" || die
+		echo "install: ;" >> "${BUILD_DIR}/Makefile" || die
+		echo "smallcheck: ;" > "${BUILD_DIR}/tests/Makefile" || die
+		return 0
+	fi
+
+	local myconf=(
+		$(use_enable "cpu_flags_x86_fma$(usex cpu_flags_x86_fma3 3 4)" fma)
+		$(use_enable fortran)
+		$(use_enable zbus mips-zbus-timer)
+		$(use_enable threads)
+		$(use_enable openmp)
+	)
+	case "${MULTIBUILD_ID}" in
+		single-*)
+			#altivec, sse, single-paired only work for single
+			myconf+=(
+				--enable-single
+				$(use_enable altivec)
```

Comment:
If I remember correctly (time to re-check), altivec only works for single precision! If it gets added for double precision configure fails, same for avx and single/long-double.